cou/apps/auxiliary.py

```diff
@@ -0,0 +1,180 @@
+from cou.utils.openstack import (
+    CHARM_TYPES,
+    LTS_SERIES,
+    AuxiliaryTrackMapping,
+    OpenStackRelease,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@AppFactory.register_application(
+    ["rabbitmq-server", "vault"] + CHARM_TYPES["mysql"] + CHARM_TYPES["ovn"]
+)
+class AuxiliaryOpenStackApplication(OpenStackApplication):
+    """Application for charms that can have multiple OpenStack releases for a workload."""
+
+    openstack_to_track_mapping = AuxiliaryTrackMapping.generate_map()
+
+    def upgrade_description(self, target: OpenStackRelease) -> str:
+        """Top description for upgrading the application.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: Description of the update.
+        :rtype: str
+        """
+        return f"Upgrade plan for '{self.name}' to {target.codename}"
+
+    def openstack_to_track(self, os_release: OpenStackRelease) -> str:
+        """Find the track of auxiliary charms by Ubuntu release and OpenStack release codename.
+
+        :param os_release: OpenStack release to track.
+        :type os_release: OpenStackRelease
+        :raises ApplicationError: When there is no track compatible.
+        :return: The track of the auxiliary charm.
+        :rtype: str
+        """
+        try:
+            return self.openstack_to_track_mapping[self.series][self.charm][os_release.codename]
+        except KeyError as exc:
+            raise ApplicationError(
+                f"Not possible to find the track for '{self.charm}' on {os_release.codename}"
+            ) from exc
+
+    def os_origin_config(self, target: OpenStackRelease) -> Optional[OpenStackRelease]:
+        """Identify the OpenStack release set on openstack-origin or source config.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :return: OpenStackRelease object or None if the app doesn't have os_origin config.
+        :rtype: Optional[OpenStackRelease]
+        """
+        os_origin_parsed = None
+        if self.os_origin is not None:
+            # Ex: "cloud:focal-ussuri" will result in "ussuri"
+            os_origin_parsed = self.os_origin.rsplit("-", maxsplit=1)[-1]
+            match os_origin_parsed:
+                case "distro":
+                    # find the OpenStack release based on ubuntu series
+                    os_origin_parsed = LTS_SERIES[self.series]
+                case "":
+                    # if it's empty we consider the previous release from the target.
+                    # Ex: rabbitmq-server has empty "source" and receive target "victoria".
+                    # In that case it will be considered as ussuri and with the upgrade,
+                    # "source" config will be changed to "cloud:focal-victoria".
+                    os_origin_parsed = target.previous_release
+        return OpenStackRelease(os_origin_parsed) if os_origin_parsed else None
```

Comment:
```suggestion
        if self.os_origin is None:
            return None

        # Ex: "cloud:focal-ussuri" will result in "ussuri"
        os_origin_parsed = self.os_origin.rsplit("-", maxsplit=1)[-1]
        if os_origin_parsed == "distro":
            # find the OpenStack release based on ubuntu series
            os_origin_parsed = LTS_SERIES[self.series]
        else:
            # if it's empty we consider the previous release from the target.
            # Ex: rabbitmq-server has empty "source" and receive target "victoria".
            # In that case it will be considered as ussuri and with the upgrade,
            # "source" config will be changed to "cloud:focal-victoria".
            os_origin_parsed = target.previous_release
        
        return OpenStackRelease(os_origin_parsed)
```